.github/scripts/trymerge.py

```diff
@@ -921,6 +921,14 @@ def gen_commit_message(self, filter_ghstack: bool = False) -> str:
         msg += f"Approved by: {approved_by_urls}\n"
         return msg
 
+    def add_numbered_label(self, label_base: str) -> None:
+        labels = self.get_labels()
+        label = label_base
+        for i in range(len(labels) if labels is not None else 0):
+            if label in labels:
+                label = f"{label_base}X{i+2}"
```

Comment:
The logic is a bit confusing, especially +2. Took me some time to convince myself it's correct.